Makefile

```diff
@@ -0,0 +1,216 @@
+# Make does not offer a recursive wildcard function, so here's one:
+rwildcard=$(wildcard $1$2) $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2))
+
+SHELL := /bin/bash
+NAME := jx-semanticcheck
+BINARY_NAME := jx-semanticcheck
+BUILD_TARGET = build
+MAIN_SRC_FILE=cmd/main.go
+GO := GO111MODULE=on go
+GO_NOMOD :=GO111MODULE=off go
+REV := $(shell git rev-parse --short HEAD 2> /dev/null || echo 'unknown')
+ORG := jenkins-x-plugins
+ORG_REPO := $(ORG)/$(NAME)
+RELEASE_ORG_REPO := $(ORG_REPO)
+ROOT_PACKAGE := github.com/$(ORG_REPO)
+GO_VERSION := 1.18
```

Comment:
Let's use 1.17.9 for now, as we are in the process of upgrading from 1.15 to 1.17.
The go version in the catalog is 1.17.9 atm.